### PR TITLE
Combine OSS, Enterprise and Commerce installation guides

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | n/a
| Versions      | 3.1, master

Companion PR to https://github.com/ezsystems/developer-documentation/pull/1147

This combines the OSS, Enterprise and Commerce installation guides into one page, plus some general cleanup.

For reviewers: the `=== "eZ Platform"` syntax is rendered as tabs, for example:
image

### CODE REVIEW

File: docs/getting_started/install_ez_platform.md [hunk 1]
```diff
@@ -149,38 +263,43 @@ The configuration requires providing the following parameters:
...
 ``` bash
 composer ezplatform-install
 ```
 
-This command will also create a database, if you had not created it earlier.
-Before executing it make sure that the database user has sufficient permissions.
+Before executing the command make sure that the database user has sufficient permissions.
 
 If Composer asks for your token, you must log in to your GitHub account and generate a new token
 (edit your profile, go to Developer settings > Personal access tokens and Generate new token with default settings).
```
Comment (dabrt) [inline docs/getting_started/install_ez_platform.md:299]:
```suggestion
(edit your profile and go to **Developer settings** > **Personal access tokens** > **Generate new token with default settings**).
```

File: docs/getting_started/install_ez_platform.md [hunk 2]
```diff
@@ -192,41 +311,42 @@ For development needs, the web user can be made the owner of all your files (for
 chown -R www-data:www-data <your installation directory>
 ```
 
-Directories `var` and `public/var` need to be writable by CLI and web server user.
+Directories `var` and `public/var` need to be writable by CLI and the web server user.
```
Comment (dabrt) [inline docs/getting_started/install_ez_platform.md:314]:
```suggestion
Directories `var` and `public/var` must be writable by CLI and the web server user.
```

File: docs/getting_started/install_ez_platform.md [hunk 3]
```diff
@@ -192,41 +311,42 @@ For development needs, the web user can be made the owner of all your files (for
...
 Future files and directories created by these two users will need to inherit those permissions.
 
 !!! caution
 
-    For security reasons, in production web server should not have write access to other directories than `var`. Skip the step above and follow the link below for production needs instead.
+    For security reasons, in production, the web server should not have write access to other directories than `var`. Skip the step above and follow the link below for production needs instead.
```
Comment (dabrt) [inline docs/getting_started/install_ez_platform.md:319]:
```suggestion
    For security reasons, in production, the web server cannot have write access to other directories than `var`. 
    Skip the step above and follow the link below for production needs instead.
```

File: docs/getting_started/install_ez_platform.md [hunk 4]
```diff
@@ -192,41 +311,42 @@ For development needs, the web user can be made the owner of all your files (for
...
 
-    You must also make sure that the web server cannot interpret files in the `var` directory through PHP. To do so, follow the instructions on [setting up a virtual host below](#set-up-virtual-host).
+    You must also make sure that the web server cannot interpret files in the `var` directory through PHP.
```
Comment (dabrt) [inline docs/getting_started/install_ez_platform.md:321]:
```suggestion
    You must also make sure that the web server cannot interpret the files in the `var` directory through PHP.
```

File: docs/getting_started/install_ez_platform.md [hunk 5]
```diff
@@ -235,21 +355,56 @@ a2ensite ezplatform
...
+## Post-installation steps
+
 !!! enterprise
 
-    If you are installing eZ Enterprise,
-    take a look at [Installing eZ Enterprise](install_ez_enterprise.md) for additional steps.
+    ### Enable Date-based Publisher
 
-!!! tip "eZ Launchpad for quick deployment"
+    To enable delayed publishing of Content using the Date-based Publisher, you need to set up cron to run the command `bin/console ezplatform:scheduled:run` periodically.
```
Comment (dabrt) [inline docs/getting_started/install_ez_platform.md:380]:
```suggestion
    To enable delayed publishing of Content using the Date-based Publisher, you must set up cron to run the `bin/console ezplatform:scheduled:run` command periodically.
```

**Review by dabrt — CHANGES_REQUESTED** [review]

File: docs/getting_started/troubleshooting.md [hunk 6]
```diff
@@ -72,3 +72,39 @@ configure the following variables in your Platform.sh environment:
 
 - `HTTPCACHE_USERNAME`
 - `HTTPCACHE_PASSWORD`
+
+## Configuration required to communicate with ERP
```
[the remark this answers is not in the document]
Comment (DominikaK) [reply docs/getting_started/troubleshooting.md:76]: Not clear yet, will be verified after reviewing the ERP section of commerce doc

File: docs/getting_started/install_ez_platform.md [hunk 7]
```diff
@@ -7,31 +7,33 @@
...
-[For production](#prepare-installation-for-production) you also need Apache or nginx as the HTTP server (Apache is used as an example below).
-[For development](#use-phps-built-in-server), you can use the built-in PHP server instead.
+- [Node.js](https://nodejs.org/en/) and [Yarn](https://yarnpkg.com/lang/en/docs/install/#debian-stable) for asset management.
+- `git` for version control.
+- to use search in the shop front end, you need to have [Solr installed](#install-solr).
 
-You also need `git` for version control.
+[For production](#prepare-installation-for-production) you also need Apache or nginx as the HTTP server (Apache is used as an example below).
 
 Before getting started, make sure you review the [requirements](requirements.md) page to see the systems we support and use for testing.
```
Comment (DominikaK) [inline docs/getting_started/install_ez_platform.md:24]: The reqs in this page are actually a subset of those on the Requirements page, so this makes more sense:
```suggestion
Before getting started, make sure you review other [requirements](requirements.md) to see the systems we support and use for testing.
```

File: docs/getting_started/install_ez_platform.md [hunk 8]
```diff
@@ -46,58 +48,164 @@ composer -V
...
+    1. Select **Create token** (this requires the **Portal administrator** access level).
+    2. Fill in a label describing the use of the token. This will allow you to revoke access later.
+    3. Save the password, **you will not get access to it again**!
 
-1\. Download an archive
+    !!! tip "Save authentication token in `auth.json` to avoid repeatedly typing it"
 
-- For open-source eZ Platform, download from [ezplatform.com](https://ezplatform.com/#download-option).
-- For licensed eZ Enterprise, download from the [Support portal](https://support.ez.no/Downloads).
+        Composer will ask to do this for you on updates.
```
Comment (DominikaK) [inline docs/getting_started/install_ez_platform.md:72]:
```suggestion
        Composer will ask you to provide a token every time you perform an update.
```

Comment (DominikaK) [inline docs/getting_started/install_ez_platform.md:72]:
```suggestion
        Composer will ask whether you want to save the token every time you perform an update.
```

File: docs/getting_started/install_ez_platform.md [hunk 9]
```diff
@@ -46,58 +48,164 @@ composer -V
...
+        If you decline, you can create an `auth.json` file manually in one of the following ways:
```
Comment (DominikaK) [inline docs/getting_started/install_ez_platform.md:73]:
```suggestion
        If you prefer, you can decline and create an `auth.json` file manually in one of the following ways:
```

File: docs/getting_started/install_ez_platform.md [hunk 10]
```diff
@@ -46,58 +48,164 @@ composer -V
...
 
-Extract the archive into the location where you want your project root directory to be.
+        - A: Store your credentials in the project directory (for security reasons do not check it in to git):
```
[the remark this answers is not in the document]
Comment (DominikaK) [reply docs/getting_started/install_ez_platform.md:75]: On this page VCS is only used as part of commands. In other places, I'd rather change VCS to git, as we're using a specific one and one less acronym is always better.

File: docs/getting_started/install_ez_platform.md [hunk 11]
```diff
@@ -46,58 +48,164 @@ composer -V
...
-Composer will look inside the `composer.json` file and install all of the packages required to run eZ Platform.
+    !!! note "Support agreement expiry"
+
+        If your Support agreement expires, your authentication token(s) will no longer work.
+        They will become active again if the agreement is renewed, but this process may take up to 24 hours.
+        _(If the agreement is renewed before the expiry date, there will be no disruption of service.)_
+
+## Create project
+
+There are two ways to get eZ Platform. The result is the same, so you can use the way you prefer:
```
Comment (DominikaK) [inline docs/getting_started/install_ez_platform.md:113]:
```suggestion
There are two ways to get an instance of eZ Platform. 
The result is the same, so you can use the way you prefer:
```

File: docs/getting_started/install_ez_platform.md [hunk 12]
```diff
@@ -46,58 +48,164 @@ composer -V
...
+
+=== "eZ Platform"
+
+    You can either:
+    - download an archive from [ezplatform.com](https://ezplatform.com/#download-option)
+    and extract the archive into the location where you want your project root directory to be, or
+    - clone the [`ezplatform` GitHub repository](https://github.com/ezsystems/ezplatform).
+
+    ``` bash
+    git clone https://github.com/ezsystems/ezplatform .
```
[the remark this answers is not in the document]
Comment (DominikaK) [reply docs/getting_started/install_ez_platform.md:128]: Cloning to the current repository avoids having to add `cd <your directory>`, which people are likely to forget.

File: docs/getting_started/install_ez_platform.md [hunk 13]
```diff
@@ -109,6 +217,11 @@ composer create-project --keep-vcs ezsystems/ezplatform .
     composer create-project --keep-vcs ezsystems/ezplatform . ^2.3.1
     ```
 
+!!! tip "Usage of authentication token with `composer create-project`"
+
+    If you have several projects set up on your machine,
```
Comment (DominikaK) [inline docs/getting_started/install_ez_platform.md:222]:
```suggestion
    If you configure several projects on one machine, make sure that
```

File: docs/getting_started/install_ez_platform.md (hunk 1 above)
Comment (DominikaK) [inline docs/getting_started/install_ez_platform.md:299]:
```suggestion
(edit your profile and go to **Developer settings** > **Personal access tokens** > **Generate new token** with default settings).
```

**Review by dabrt — APPROVED** [review]

**Pull request merged**